lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -530,7 +530,114 @@ swift::matchWitness(
   }
 
   // Now finalize the match.
-  return finalize(anyRenaming, optionalAdjustments);
+  auto result = finalize(anyRenaming, optionalAdjustments);
```

Comment:
What do you think about adding this separately, and just forbidding @diffble requirements on master for now? It would allow you to make this PR smaller (also you'll be able to exclude the whole derivative function configuration thing), and would let the code owners review this as an independent patch where we give more context about how it works and what it enables.

Reply:
I'm open to splitting this PR, but I won't do so preemptively. I plan to cc a code owner soon - we can split if requested.